### DISCUSSION TITLE
Incorrect resize

### DISCUSSION THREAD

**Discussion by Sepera-okeq** [ImageMagick/ImageMagick#3071] [Help] [answered]
Hi, there's a problem. I would like to know why resize behaves so strangely.
I want to convert a bunch of files to a certain size 10x7.
I write like this:
``magick mogrify -path D:\Users\User\Documents\small -resize 10x7 D:\Users\User\Documents\normal\*.tga``

But I get 10x6 in the end, why so?

OS: Windows 10
Version: ImageMagick 7.0.10-54 Q16 x64

**Comment by fmw42** [comment]
Please post an example input TGA file.

**Comment by Sepera-okeq** [comment]
Here you go.
[tga.zip](https://github.com/ImageMagick/ImageMagick/files/5762173/tga.zip)

**Comment by snibgo** [comment, marked as answer]
> -resize 10x7

IM tries to maintain the aspect ratio. If you want _exactly_ 10x7 pixels, append "!" like this:
```
-resize 10x7!
```

**Comment by fmw42** [comment]
Your input is 82x52 and you want 10x7 resize. But the aspect ratios are not the same, the resize width should be 82/10=8.2 rounded to 8 and the resize height should be 52/7 = 7.428571428571429 and the rounded height should be 7. I am not sure why the TGA resize is 8x6. It should properly be 8x7. **So there is some kind of rounding issue.** But you can use -resize 10x7! to force an exect size.

**Comment by Sepera-okeq** [comment]
Thanks for the help.
